### PR TITLE
Business logic to decide and choose the Bus API to use for BLOB SET

### PR DESCRIPTION
**Pull request by Sreedevi29**

### CODE REVIEW

File: src/webcfg_event.c [hunk 1]
```diff
@@ -969,7 +971,7 @@ WEBCFG_STATUS retryMultipartSubdoc(webconfig_tmp_data_t *docNode, char *docName)
 						{
 							char *appended_doc = NULL;
 							WebcfgDebug("B4 webcfg_appendeddoc\n");
-							appended_doc = webcfg_appendeddoc( gmp->name_space, gmp->etag, pm->entries[i].value, pm->entries[i].value_size, &doc_transId);
+							appended_doc = webcfg_appendeddoc( gmp->name_space, gmp->etag, pm->entries[i].value, pm->entries[i].value_size, &doc_transId, &sendMsgsize);
```
Comment (sadhyama) [inline src/webcfg_event.c:974]: Business logic need to be added during blob retry also here,  currently setValues is used during retry cases. ,
https://github.com/xmidt-org/webcfg/blob/22fa82854e91ecf3d0b5b3feace066bd8d190753/src/webcfg_event.c#L1005

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: src/webcfg_multipart.c [hunk 2]
```diff
@@ -563,6 +568,11 @@ WEBCFG_STATUS processMsgpackSubdoc(char *transaction_id)
 	int current_doc_count = 0;
 	int err = 0;
 	char * errmsg = NULL;
+
+	int sendMsgSize = 0;
+	#ifdef WEBCONFIG_BIN_SUPPORT
+		void *buff = NULL;
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:574]: Move this declaration after line 685 where ifdef is defined

**Review by sadhyama — APPROVED** [review]

File: src/webcfg_event.c [hunk 3]
```diff
@@ -969,13 +978,30 @@ WEBCFG_STATUS retryMultipartSubdoc(webconfig_tmp_data_t *docNode, char *docName)
...
+							
 							WebcfgDebug("webcfg_appendeddoc doc_transId is %hu\n", doc_transId);
 							reqParam[i].name = strdup(pm->entries[i].name);
 							WebcfgDebug("appended_doc length: %zu\n", strlen(appended_doc));
```
Comment (shilpa24balaji) [inline src/webcfg_event.c:985]: When rbus listener support is true strlen on binary data will not give correct results

File: src/webcfg_event.c [hunk 4]
```diff
@@ -969,13 +978,30 @@ WEBCFG_STATUS retryMultipartSubdoc(webconfig_tmp_data_t *docNode, char *docName)
...
-							reqParam[i].value = strdup(appended_doc);
-							reqParam[i].type = WDMP_BASE64;
-							WEBCFG_FREE(appended_doc);
+							#ifdef WEBCONFIG_BIN_SUPPORT
+								if(isRbusEnabled() && isRbusListener(gmp->name_space))
+								{
+								        //disable string operation as it is binary data.
+									reqParam[i].value = appended_doc;
+									reqParam[i].type = WDMP_BASE64;
+									buff = reqParam[i].value;
```
Comment (shilpa24balaji) [inline src/webcfg_event.c:992]: reqParam set is not needed in this case. buff can be just set to appended_doc

File: src/webcfg_multipart.c [hunk 5]
```diff
@@ -672,9 +681,25 @@ WEBCFG_STATUS processMsgpackSubdoc(char *transaction_id)
...
-							reqParam[i].value = strdup(appended_doc);
-							reqParam[i].type = WDMP_BASE64;
-							WEBCFG_FREE(appended_doc);
+							#ifdef WEBCONFIG_BIN_SUPPORT
+								if(isRbusEnabled() && isRbusListener(mp->name_space))
+								{
+								        //disable string operation as it is binary data.
+									reqParam[i].value = appended_doc;
+									reqParam[i].type = WDMP_BASE64;
+									buff = reqParam[i].value;
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:690]: I feel reqParam set is not needed here

File: src/webcfg_metadata.c [hunk 6]
```diff
@@ -256,6 +278,59 @@ WEBCFG_STATUS isSubDocSupported(char *subDoc)
...
+	SubDocSupportMap_t *sd = NULL;
+
+	sd = get_global_sdInfoHead();
+
+	while(sd != NULL)
+	{
+		if(strncmp(sd->name, subDoc, strlen(subDoc)) == 0)
+		{
+			WebcfgDebug("The subdoc %s is present\n",sd->name);
+			if(strncmp(sd->rbus_listener, "true", strlen("true")) == 0)
```
Comment (shilpa24balaji) [inline src/webcfg_metadata.c:295]: add null check for rbus_listener

File: src/webcfg_metadata.c [hunk 7]
```diff
@@ -256,6 +278,59 @@ WEBCFG_STATUS isSubDocSupported(char *subDoc)
...
+WEBCFG_STATUS get_destination(char* subDoc, char* destination)
+{
+	SubDocSupportMap_t *sd = NULL;
+	sd = get_global_sdInfoHead();
+
+	while(sd != NULL)
+	{
+		if(strncmp(sd->name, subDoc, strlen(subDoc)) == 0)
+		{
+			if( sd->dest != NULL)
```
Comment (shilpa24balaji) [inline src/webcfg_metadata.c:323]: dest is char[] so NULL check is not applicable

**Pull request merged**